### PR TITLE
fix(les_fvm): open-wall BC consistency — cell-Dirichlet inflow + periodic seams (closes #60)

### LINKED ISSUE #60: les_fvm open-wall BC consistency: cell-Dirichlet advective inflow + periodic halo wrapping (wind & field K_h)
Follow-up to #59 (which wired `les_fvm` horizontal advection/diffusion to finitevolX's `wall="open"` mode, closing #25/#26). Three related correctness refinements surfaced in review were deferred from that PR because each needs a design change beyond the wiring, and all are low-impact for the shipped default (scalar `K_h`, zero-Dirichlet west, downwind plume). Root theme: **open-mode operators read the ghost ring, so every per-axis BC must be reflected consistently in the tracer *and* the coefficient/velocity halos.**

### 1. Cell-Dirichlet ghost for advective inflow (P1)
`apply_boundary_conditions` fills a Dirichlet ghost with the *face* reflection `2·value − interior` (correct for the diffusion centred gradient). Open-mode advection reads the ghost as the exterior *cell* value, so a Dirichlet **inflow** carries `2·value − interior` instead of `value` — for a zero-Dirichlet inflow with tracer present at the wall this injects negative concentration.

- Fix: give advection a **cell-Dirichlet** ghost (`ghost = value`) while diffusion keeps the face reflection — a separate advective BC fill applied to the advection input in `dynamics.py` (`boundary.py` gains an advection-oriented variant).
- Upstream context: jejjohnson/finitevolX#235 (the operator can't resolve this without breaking periodic conservation; it's a consumer-side ghost-fill choice).

### 2. Wrap normal velocity at periodic seams (P2)
For a periodic axis with spatially-varying normal wind, `PrescribedWindField` edge-pads the velocity halo while the tracer is wrapped, so `u[...,0]` ≠ `u[...,-2]` and the two paired seam faces give different fluxes → mass not conserved.

- Fix: wrap the normal-velocity halo on periodic axes (same treatment as the tracer), in the RHS where the periodicity of each axis is known.

### 3. Wrap field diffusivity at periodic seams (P2)
Same mechanism for a field `EddyDiffusivity.horizontal`: `as_arrays()` edge-pads `K_h`, so a periodic diffusive seam uses different coefficients on its paired faces and stops cancelling. (Scalar `K_h` — the default — conserves unconditionally.)

- Fix: fill the `K_h` halo consistently with the tracer BC (periodic → wrapped) on periodic axes.

### Definition of done
- [ ] Advective Dirichlet inflow carries the boundary value (budget test with nonzero near-wall tracer).
- [ ] Periodic-axis conservation holds with a spatially-varying normal wind (test with non-uniform periodic wind).
- [ ] Periodic-axis conservation holds with a field `K_h` (test with non-uniform periodic `K_h`).
- [ ] `make test && make lint && make typecheck` green.

### Relationships
- Parent epic: #19
- Follows: #59

_Filed by [Claude Code](https://claude.ai/code) while landing #59._

### PR DESCRIPTION
**Pull request by jejjohnson**
## Summary

Closes #60 — the three open-wall BC-consistency refinements deferred from #59. Open-mode advection/diffusion read the ghost ring, so a per-axis BC has to be reflected consistently in the tracer **and** the coefficient/velocity halos. All three were low-impact for the shipped default (scalar `K_h`, downwind plume) but are real correctness gaps for other configs.

## Changes

**1. Cell-Dirichlet ghost for advective inflow (was P1)**
`apply_boundary_conditions` fills a Dirichlet ghost with the *face* reflection `2·value − interior` (right for diffusion's centred gradient). Open-mode advection upwinds the ghost as an exterior *cell* value, so an inflow carried `2·value − interior` instead of `value` — a zero-Dirichlet inflow with near-wall tracer could inject negative concentration.
- New `CellDirichlet1D` atom (`ghost = value`) and `HorizontalBC.advection_variant()` / `VerticalBC.advection_variant()`.
- `EulerianDispersionRHS` now fills the ghost **twice** — cell-Dirichlet for advection, face-Dirichlet for diffusion — differing only at Dirichlet faces (outflow/periodic/Neumann are identical, so it's a no-op otherwise).

**2. Periodic wind seam (was P2)**
On a periodic axis the normal-velocity halo was edge-padded while the tracer wrapped, so a spatially-varying wind gave the two seam faces different velocities and broke conservation. `advection_tendency` gains `periodic=(x,y)` and wraps the normal-velocity halo on periodic axes.

**3. Periodic field-`K_h` seam (was P2)**
Same for a spatially-varying `EddyDiffusivity.horizontal`: `diffusion_tendency` gains `periodic=(x,y)` and wraps the `K_h` halo (a scalar `K_h` is uniform and already conserves). The RHS detects periodic axes with `boundary.periodic_axes` and threads the flag through both tendencies.

## Tests

`tests/les_fvm/test_open_wall_bc_consistency.py`:
- cell- vs face-Dirichlet ghost (horizontal **and** vertical faces);
- no negative injection at a zero-Dirichlet inflow over 30 steps;
- periodic wind-seam and field-`K_h`-seam conservation under **non-uniform** fields (total interior tendency → 0 wrapped, ≠ 0 unwrapped);
- scalar-`K_h` conserves unconditionally; per-axis wrapping; RHS integration (finite, interior-only).

Full `les_fvm` suite green: **124 passed**. `ruff check .` / `ruff format --check .` / `ty check src/plumax` all pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01E3NxHCEGifJAEEu8kjKNKb)_

### CODE REVIEW

File: src/plumax/les_fvm/boundary.py [hunk 1]
```diff
@@ -162,6 +254,25 @@ def apply_boundary_conditions(
...
+    two representations of a periodic seam face share a coefficient and the
+    open-wall fluxes cancel.
+    """
+    bc_set = horizontal_bc.bc_set
+    x_periodic = isinstance(bc_set.west, Periodic1D) or isinstance(
+        bc_set.east, Periodic1D
+    )
+    y_periodic = isinstance(bc_set.south, Periodic1D) or isinstance(
+        bc_set.north, Periodic1D
+    )
```
[the remark this answers is not in the document]
Comment (jejjohnson) [reply src/plumax/les_fvm/boundary.py]: Fixed in `08c35bb` by rejecting the mixed config. Periodicity is an axis property — the seam couples both faces — so a half-periodic axis like `bc_x=(("periodic",0),("dirichlet",0))` is ill-defined, and per-face wrapping would still leave the seam inconsistent. `periodic_axes` now raises `ValueError` when exactly one face of an axis is `Periodic1D` (both-or-neither), with a regression test. Fully-periodic and fully-non-periodic axes are unaffected.

---
_Generated by [Claude Code](https://claude.ai/code)_
